app/controllers/concerns/foreman/controller/authentication.rb

```diff
@@ -71,14 +71,17 @@ def sso_authentication
 
   def set_current_user(user)
     User.current = user
-
     # API access resets the whole session and marks the session as initialized from API
     # such sessions aren't checked for CSRF
     # UI access resets only session ID
     if api_request?
+      jwt_data = { sso_method: session[:sso_method], expires_at: session[:expires_at] }
       reset_session
       session[:user] = user.id
       session[:api_authenticated_session] = true
+      if jwt_data[:sso_method] == "SSO::Jwt"
+        add_jwt_sesssion_values(jwt_data)
```

Comment:
How about `session.merge!(jwt_data)`

Reply:
Doesn't this store data in the session that was read from the session a couple of lines before? 😕

Reply:
It should have, but since we have the `reset_session` on line no.79, it resets all the values and therefore I had to  use this method to restore the value again.